pkg/engine/engine.go

```diff
@@ -208,6 +232,9 @@ func (e *Engine) DetectorAvgTime() map[string][]time.Duration {
 func (e *Engine) detectorWorker(ctx context.Context) {
 	for originalChunk := range e.chunks {
 		for chunk := range sources.Chunker(originalChunk) {
+
+			chunk.Keywords = make(map[string]bool)
```

Comment:
I'm not sure if it makes as much differences these days, or for this use case, but using struct{} instead of bool has a measurable impact on CPU and memory performance: https://itnext.io/set-in-go-map-bool-and-map-struct-performance-comparison-5315b4b107b